From e508439f98b51ca8ce0288188211e187ba3ed41a Mon Sep 17 00:00:00 2001
From: JeongBong Seo <jb.seo@lge.com>
Date: Wed, 9 Feb 2022 12:00:57 +0900
Subject: [PATCH] Speed up boot by disable usb and enable bootcount

PREBOOT was moved from rpi.h to Kconfig in v2019.10-rc2:
commit 37304aaf60bf92a5dc3ef222ba520698bd862a44
Author: Simon Glass <sjg@chromium.org>
Date:   Sat Jul 20 20:51:14 2019 -0600

  Convert CONFIG_USE_PREBOOT and CONFIG_PREBOOT to Kconfig

DISTRO_DEFAULTS was removed in:
90f626a5d8 rpi: Disable DISTRO_DEFAULTS

environment moved from include/configs/rpi.h to board/raspberrypi/rpi/rpi.env in:
fc2af2d978 arm: rpi: Switch to a text environment
---
Upstream-Status: Pending

 board/raspberrypi/rpi/rpi.env  | 6 +++---
 configs/rpi_3_32b_defconfig    | 5 +++++
 configs/rpi_3_b_plus_defconfig | 5 +++++
 configs/rpi_3_defconfig        | 5 +++++
 configs/rpi_4_32b_defconfig    | 5 +++++
 configs/rpi_4_defconfig        | 5 +++++
 configs/rpi_arm64_defconfig    | 5 +++++
 7 files changed, 33 insertions(+), 3 deletions(-)

diff --git a/board/raspberrypi/rpi/rpi.env b/board/raspberrypi/rpi/rpi.env
index 30228285ed..cb2ff7954d 100644
--- a/board/raspberrypi/rpi/rpi.env
+++ b/board/raspberrypi/rpi/rpi.env
@@ -5,9 +5,9 @@
 dhcpuboot=usb start; dhcp u-boot.uimg; bootm
 
 /* Environment */
-stdin=serial,usbkbd
-stdout=serial,vidconsole
-stderr=serial,vidconsole
+stdin=serial
+stdout=serial
+stderr=serial
 
 /* DFU over USB/UDC */
 #ifdef CONFIG_CMD_DFU
diff --git a/configs/rpi_3_32b_defconfig b/configs/rpi_3_32b_defconfig
index 0ccc93d572..c4c8691d2a 100644
--- a/configs/rpi_3_32b_defconfig
+++ b/configs/rpi_3_32b_defconfig
@@ -51,3 +51,8 @@ CONFIG_SYS_WHITE_ON_BLACK=y
 CONFIG_VIDEO_BCM2835=y
 CONFIG_CONSOLE_SCROLL_LINES=10
 CONFIG_PHYS_TO_BUS=y
+
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
+
+CONFIG_BOOTDELAY=0
diff --git a/configs/rpi_3_b_plus_defconfig b/configs/rpi_3_b_plus_defconfig
index 53f7f3117c..428452cf07 100644
--- a/configs/rpi_3_b_plus_defconfig
+++ b/configs/rpi_3_b_plus_defconfig
@@ -50,3 +50,8 @@ CONFIG_SYS_WHITE_ON_BLACK=y
 CONFIG_VIDEO_BCM2835=y
 CONFIG_CONSOLE_SCROLL_LINES=10
 CONFIG_PHYS_TO_BUS=y
+
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
+
+CONFIG_BOOTDELAY=0
diff --git a/configs/rpi_3_defconfig b/configs/rpi_3_defconfig
index 84b61f862a..4d72994aa6 100644
--- a/configs/rpi_3_defconfig
+++ b/configs/rpi_3_defconfig
@@ -50,3 +50,8 @@ CONFIG_SYS_WHITE_ON_BLACK=y
 CONFIG_VIDEO_BCM2835=y
 CONFIG_CONSOLE_SCROLL_LINES=10
 CONFIG_PHYS_TO_BUS=y
+
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
+
+CONFIG_BOOTDELAY=0
diff --git a/configs/rpi_4_32b_defconfig b/configs/rpi_4_32b_defconfig
index 766c983196..1fd3c00c83 100644
--- a/configs/rpi_4_32b_defconfig
+++ b/configs/rpi_4_32b_defconfig
@@ -67,3 +67,8 @@ CONFIG_CONSOLE_SCROLL_LINES=10
 CONFIG_PHYS_TO_BUS=y
 CONFIG_ADDR_MAP=y
 CONFIG_SYS_NUM_ADDR_MAP=2
+
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
+
+CONFIG_BOOTDELAY=0
diff --git a/configs/rpi_4_defconfig b/configs/rpi_4_defconfig
index 553d47b174..a8bf283a4a 100644
--- a/configs/rpi_4_defconfig
+++ b/configs/rpi_4_defconfig
@@ -65,3 +65,8 @@ CONFIG_SYS_WHITE_ON_BLACK=y
 CONFIG_VIDEO_BCM2835=y
 CONFIG_CONSOLE_SCROLL_LINES=10
 CONFIG_PHYS_TO_BUS=y
+
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
+
+CONFIG_BOOTDELAY=0
diff --git a/configs/rpi_arm64_defconfig b/configs/rpi_arm64_defconfig
index 7ff166a41d..9208ae637a 100644
--- a/configs/rpi_arm64_defconfig
+++ b/configs/rpi_arm64_defconfig
@@ -57,3 +57,8 @@ CONFIG_SYS_WHITE_ON_BLACK=y
 CONFIG_VIDEO_BCM2835=y
 CONFIG_CONSOLE_SCROLL_LINES=10
 CONFIG_PHYS_TO_BUS=y
+
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
+
+CONFIG_BOOTDELAY=0
